### ISSUE TITLE
`<format> + <chrono>`: Could not format time_points

### ISSUE THREAD

**Issue by nico-engels** [StephanTLavavej/STL#16]
# Basic Info
* [x] I have read the rules for the [Standard Library Modules Bug Bash][bug-bash-url].

* [x] I have pulled and rebuilt the `bug-bash` branch in the last 24 hours.

* My current commit, as printed by `git rev-parse HEAD`, is: 6c96fe702ab23df846cca2110bc8dc08e2aa6d3a

* This bug is:
rejects-valid

# Command-Line Test Case
My test case that works with classic includes but fails with named modules:

```
d:\cpp\STL\cpp-meo>type bug-2.cpp
```
```cpp
#ifdef USE_CLASSIC_INCLUDES
#include <chrono>
#include <format>
#include <iostream>
#else
import std;
#endif

int main()
{
    std::cout << std::format("{}\n", std::chrono::system_clock::time_point {});
}
```
```
d:\cpp\STL\cpp-meo>cl /EHsc /nologo /W4 /std:c++latest /reference ..\std.ifc /MTd /Od bug-2.cpp ..\std.obj /DUSE_CLASSIC_INCLUDES && bug-2.exe/DUSE_CLASSIC_INCLUDES && bug-1.exe
bug-2.cpp
1970-01-01 00:00:00.0000000

d:\cpp\STL\cpp-meo>cl /EHsc /nologo /W4 /std:c++latest /reference ..\std.ifc /MTd /Od bug-2.cpp ..\std.obj && bug-2.exe
bug-2.cpp
d:\cpp\STL\out\x64\out\inc\chrono(5827): error C3861: '_Choose_literal': identificador não encontrado
d:\cpp\STL\out\x64\out\inc\chrono(5828): note: Este diagnóstico ocorreu na função gerada pelo compilador 'std::formatter<std::chrono::time_point<std::chrono::system_clock,std::chrono::duration<__int64,std::ratio<1,10000000>>>,_CharT>::formatter(void)'
        with
        [
            _CharT=char
        ]
d:\cpp\STL\out\x64\out\inc\format(3176): note: confira a referência à função 'std::formatter<std::chrono::time_point<std::chrono::system_clock,std::chrono::duration<__int64,std::ratio<1,10000000>>>,_CharT>::formatter(void)'
        with
        [
            _CharT=char
        ]
d:\cpp\STL\out\x64\out\inc\format(3190): note: consulte a referência à instanciação 'std::_String_view_iterator<_Traits> std::_Compile_time_parse_format_specs<std::chrono::time_point<std::chrono::system_clock,std::chrono::duration<__int64,std::ratio<1,10000000>>>,std::basic_format_parse_context<_CharT>>(_ParseContext &)' do modelo que está sendo compilada
        with
        [
            _Traits=std::char_traits<char>,
            _CharT=char,
            _ParseContext=std::basic_format_parse_context<char>
        ]
d:\cpp\STL\out\x64\out\inc\format(3190): note: ao compilar a função de membro 'std::_Format_checker<_CharT,std::chrono::time_point<std::chrono::system_clock,std::chrono::duration<__int64,std::ratio<1,10000000>>>>::_Format_checker(std::basic_string_view<_CharT,std::char_traits<char>>) noexcept' do modelo da classe
        with
        [
            _CharT=char
        ]
d:\cpp\STL\out\x64\out\inc\format(3355): note: consulte a referência à instanciação 'std::_Format_checker<_CharT,std::chrono::time_point<std::chrono::system_clock,std::chrono::duration<__int64,std::ratio<1,10000000>>>>' do modelo da classe que está sendo compilada
        with
        [
            _CharT=char
        ]
bug-2.cpp(13): note: consulte a referência à instanciação 'std::_Basic_format_string<char,std::chrono::time_point<std::chrono::system_clock,std::chrono::duration<__int64,std::ratio<1,10000000>>>>::_Basic_format_string<char[4]>(const _Ty (&))' do modelo que está sendo compilada
        with
        [
            _Ty=char [4]
        ]
```

# Additional Context
Based on #14, but a diferent error.

[bug-bash-url]: https://github.com/StephanTLavavej/STL/wiki/Standard-Library-Modules-Bug-Bash

**Comment by StephanTLavavej** [comment]
Thanks for finding this! I'll investigate later today - I believe the workaround that I applied for #14 will be applicable here too.

**Comment by StephanTLavavej** [comment]
Yep, looks like the same root cause. I've pushed a workaround that unblocks this test case.

Thanks again! :heart_eyes_cat:

**Comment by StephanTLavavej** [comment]
Upon further reflection, I think this may have a different root cause - the compiler error pointed to `_Choose_literal` being used within a Non-Static Data Member Initializer:

https://github.com/StephanTLavavej/STL/blob/e7b01974b8835b033891d926ff8a29eec10bc714/stl/inc/chrono#L4643
https://github.com/StephanTLavavej/STL/blob/e7b01974b8835b033891d926ff8a29eec10bc714/stl/inc/chrono#L5827

That would be internal [VSO-1538666](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1538666) "Compiler does not bind NSDMIs inside class templates", which has already been fixed for 17.4 Preview 2. I'll mark the library workaround appropriately.
